test/controllers/assignments_controller_test.rb

```diff
@@ -812,4 +812,102 @@ class AssignmentsControllerTest < ActionController::TestCase
   end
 
+  ##################################################
+  # SubmissionEnabledToggles and interlocks
+  ##################################################
+  test "should not allow two check_section_toggles interlocks on create" do
+    sign_in(@fred)
+    post :create, params: {
+      course_id: @cs101.id,
+      assignment: {
+        teamset_plan: "none",
+        assignment: "Dance a jig.",
+        points_available: 100,
+        name: "Useful Work",
+        due_date: '2019-05-22',
+        available: '2011-05-22',
+        type: "Files",
+        graders_attributes: {
+          "1477181088065"=> {
+            "type"=>"ManualGrader",
+            "avail_score"=>"50",
+            "order"=>"1",
+            "_destroy"=>"false",
+          }
+        },
+        interlocks_attributes: {
+          "1539721230721" => {
+            "constraint"=>"check_section_toggles"
+          },
+          "1539721230722" => {
+            "constraint"=>"check_section_toggles"
+          }
+        },
+        lateness_config_attributes: {
+          "type"=>"LatePerHourConfig",
+          "frequency"=>"1",
+          "percent_off"=>"25",
+          "max_penalty"=>"100",
+          "days_per_assignment"=>"22"
+        },
+      }
+    }
+    assert_response 400
+    @assn = assigns(:assignment)
+    assert_equal(["Can only have one section-based interlock"], @assn.errors.full_messages)
+  end
+
+  test "should not allow two check_section_toggles interlocks on update" do
+    sign_in(@fred)
+    post :create, params: {
+      course_id: @cs101.id,
+      assignment: {
+        teamset_plan: "none",
+        assignment: "Dance a jig.",
+        points_available: 100,
+        name: "Useful Work",
+        due_date: '2019-05-22',
+        available: '2011-05-22',
+        type: "Files",
+        graders_attributes: {
+          "1477181088065"=> {
+            "type"=>"ManualGrader",
+            "avail_score"=>"50",
+            "order"=>"1",
+            "_destroy"=>"false",
+          }
+        },
+        interlocks_attributes: {
+          "1539721230721" => {
+            "constraint"=>"check_section_toggles"
+          }
+        },
+        lateness_config_attributes: {
+          "type"=>"LatePerHourConfig",
+          "frequency"=>"1",
+          "percent_off"=>"25",
+          "max_penalty"=>"100",
+          "days_per_assignment"=>"22"
+        },
+      }
+    }
+    assert_response 302
+    @assn = assigns(:assignment).becomes(Assignment)
+    assert_redirected_to [@cs101, @assn]
+
+    put :update, params: {
+      id: @assn.id,
+      course_id: @cs101.id,
+      assignment: {
+        interlocks_attributes: {
+          "1539721230722" => {
+            "constraint"=>"check_section_toggles"
+          }
+        },
+      }
+    }
+    assert_response 400
+    @assn = assigns(:assignment)
+    assert_equal(["Can only have one section-based interlock"], @assn.errors.full_messages)
+  end
```

Comment:
@blerner please let me know if these tests look good. Just want to make sure I am writing tests correctly. They pass :smile:

Reply:
That's a good start.  Now we need tests that creating *one* such toggle *does* work, and then some submissions_controller tests that check whether the toggles have the intended effects...  I would split up your second (update) test into two tests -- one that successfully uses controller methods to create an assignment with a toggle, and a second one that attempts to use controller methods to update an existing assignment (which you'd create directly using model constructors).